### PR TITLE
Add 'sandbox-pod' command. Implement test.

### PR DESCRIPTION
**Pull request by e2l3n**
Add `sandbox-pod` command execution to restirct write permissions outside project directory.
TeamPulse - [[Server] Sandbox Cocoa Pods](http://teampulse.telerik.com/view#item/302897)
Related: https://github.com/NativeScript/nativescript-cli/issues/1065

### CODE REVIEW

**Comment by e2l3n** [comment]
Is it possible that `CocoaPods` is not installed on our build machine?

File: test/project-service.ts [hunk 1]
```diff
@@ -147,12 +145,34 @@ describe("Project Service Tests", () => {
...
 
 			projectIntegrationTest.createProject(projectName).wait();
 			projectIntegrationTest.assertProject(tempFolder, projectName, options.appid).wait();
 		});
+		it("creates valid project and tests pod sandbox", () => {
+			let testInjector = createTestInjector();
+			let fs: IFileSystem = testInjector.resolve("fs");
+			let config = testInjector.resolve("config");
+			let childProcess = testInjector.resolve("childProcess");
+			let	projectIntegrationTest = new ProjectIntegrationTest();
```
Comment (teobugslayer) [inline test/project-service.ts]: Tab <-> spaces issue? there's extra white space

Comment (e2l3n) [inline test/project-service.ts]: Fixed incorrect spacing.

**Comment by e2l3n** [comment]
Regarding the CI error message 
```
Error: spawn ENOENT
    at Object.Future.wait (/home/travis/build/NativeScript/nativescript-cli/node_modules/fibers/future.js:535:15)
    at Context.<anonymous> (/home/travis/build/NativeScript/nativescript-cli/test/project-service.js:142:118)
    at /home/travis/build/NativeScript/nativescript-cli/node_modules/mocha-fibers/lib/mocha-fibers.js:25:14
    - - - - -
    at errnoException (child_process.js:1011:11)
    at Process.ChildProcess._handle.onexit (child_process.js:802:34)
```
Line 142 corresponds to 
```
childProcess.spawnFromEvent(podTool, ["install"], "close", { cwd: workingFolderPath, stdio: 'inherit' }).wait();
```

I strongly believe the build machine doesn't have `CocoaPods` installed.

File: test/project-service.ts [hunk 2]
```diff
@@ -39,21 +39,20 @@ class ProjectIntegrationTest {
 		return projectService.createProject(projectName);
 	}
 
-	public getDefaultTemplatePath(): IFuture<string> {
+	public getDefaultTemplatePath(templateName: string): IFuture<string> {
```
Comment (Fatme) [inline test/project-service.ts]: This function should be renamed for example `getNpmPackagePath(packageName: string)`

File: test/project-service.ts [hunk 3]
```diff
@@ -147,12 +145,39 @@ describe("Project Service Tests", () => {
...
+			let iosTemplatePath = path.join(projectIntegrationTest.getDefaultTemplatePath("tns-ios").wait(), "framework/");
+			childProcess.exec(`cp -R ${iosTemplatePath} ${workingFolderPath}`, { cwd: workingFolderPath }).wait();
+			fs.writeFile("/tmp/Podfile/testFile.txt", "Test content.").wait();
+
+			let postInstallCommmand = `\`cat /tmp/Podfile/testFile.txt > ${workingFolderPath}/copyTestFile.txt && rm -rf /tmp/Podfile\``;
+			let podfileContent = `post_install do |installer_representation| ${postInstallCommmand} end`;
+			fs.writeFile(path.join(workingFolderPath, "Podfile"), podfileContent).wait();
+
+			let podTool = config.USE_POD_SANDBOX ? "sandbox-pod" : "pod";
+			childProcess.spawnFromEvent(podTool,  ["install"], "close", { cwd: workingFolderPath, stdio: 'inherit' }).wait();
```
Comment (Fatme) [inline test/project-service.ts]: Shouldn't we call here `executePodInstall` function from `iosProjectService`? This way if we break something the test will fail.

**Comment by Fatme** [comment]
:+1:  after  taking care of my comments.

File: test/project-service.ts [hunk 4]
```diff
@@ -147,12 +146,52 @@ describe("Project Service Tests", () => {
...
+			let testInjector = createInjectorForPodsTest();
+
+			let iOSProjectService: IPlatformProjectService = testInjector.resolve("iOSProjectService");
+			let fs: IFileSystem = testInjector.resolve("fs");
+			let childProcess = testInjector.resolve("childProcess");
+			let projectIntegrationTest = new ProjectIntegrationTest();
+			let workingFolderPath = temp.mkdirSync("ios_project");
+
+			let iosTemplatePath = path.join(projectIntegrationTest.getDefaultTemplatePath("tns-ios").wait(), "framework/");
+			childProcess.exec(`cp -R ${iosTemplatePath} ${workingFolderPath}`, { cwd: workingFolderPath }).wait();
```
Comment (teobugslayer) [inline test/project-service.ts]: You should use either $fs or shelljs

File: test/project-service.ts [hunk 5]
```diff
@@ -147,12 +146,52 @@ describe("Project Service Tests", () => {
...
+			fs.writeFile("/tmp/Podfile/testFile.txt", "Test content.").wait();
+
+			let postInstallCommmand = `\`cat /tmp/Podfile/testFile.txt > ${workingFolderPath}/copyTestFile.txt && rm -rf /tmp/Podfile\``;
```
Comment (teobugslayer) [inline test/project-service.ts]: magic constant used more than once - consider making it a symbol

**Comment by e2l3n** [comment]
Changes have been applied. Please review again. The timeout has been increased to 15 seconds but sporadically the test fails with a `timeout` error. This should happen very rarely though but weshould have into account.

**Comment by rosen-vladimirov** [comment]
:+1:

**Pull request merged**